agr_literature_service/lit_processing/oneoff_scripts/update_reference_curie.py

```diff
@@ -0,0 +1,81 @@
+import logging
+import requests
+from os import environ
+
+from agr_literature_service.lit_processing.utils.sqlalchemy_utils import create_postgres_session
+from agr_literature_service.lit_processing.utils.okta_utils import (
+    get_authentication_token,
+    generate_headers
+)
+
+logging.basicConfig(format='%(message)s')
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def update_curies():
+
+    db_session = create_postgres_session(False)
+
+    limit = 500
+    loop_count = 2500
+
+    for index in range(loop_count):
+
+        offset = index * limit
+
+        rs = db_session.execute("SELECT reference_id, curie FROM reference order by reference_id limit " + str(limit) + " offset " + str(offset))
+        rows = rs.fetchall()
+
+        if len(rows) == 0:
+            break
+
+        curie_count = len(rows)
+        (curr_curie, counter) = get_first_curie(curie_count)
+
+        for x in rows:
+            reference_id = x[0]
+            old_curie = x[1]
+            new_curie = curr_curie
+
+            db_session.execute("UPDATE reference set curie = '" + new_curie + "' WHERE reference_id = " + str(reference_id))
+
+            curr_curie = get_next_curie(curr_curie, counter)
+            counter += 1
+            logger.info(str(reference_id) + " " + old_curie + " " + new_curie)
+
+        db_session.commit()
+
+    db_session.close()
+
+
+def get_next_curie(curie, counter):
+
+    curie_len = len(curie)
+    counter_len = len(str(counter))
+    next_curie = curie[0:curie_len - counter_len] + str(counter + 1)
```

Comment:
@sweng66 I just want to check I understand this.  get_first_curie returns res_json['first']['curie'] which is a curie string, and that gets passed here as `curie`.  so curie_len is the amount of characters in the string, and counter_len is the amount of characters in the counter number.  Then `curie[0:curie_len - counter_len]` splices the curie string to have the same beginning, and then appends counter+1, which could in theory be one digit, and then 10 curies later it would be two digits, and it works out because the counter variable is getting updated in the update_curies for loop ?

Reply:
yes, correct!

Reply:
Thanks @sweng66 !